src/main/java/io/cryostat/configuration/CredentialsManager.java

```diff
@@ -204,11 +226,86 @@ public Collection<ServiceRef> getServiceRefsWithCredentials() {
+        }
+        try (BufferedReader br = fs.readFile(path)) {
+            StoredCredentials sc = gson.fromJson(br, StoredCredentials.class);
+            return sc.getMatchExpression();
+        }
+    }
+
+    public Set<ServiceRef> resolveMatchingTargets(int id) throws IOException {
+        Path path = credentialsDir.resolve(String.valueOf(id));
+        try (BufferedReader br = fs.readFile(path)) {
+            StoredCredentials sc = gson.fromJson(br, StoredCredentials.class);
+            Set<ServiceRef> matchedTargets = new HashSet<>();
+            for (ServiceRef target : platformClient.listDiscoverableServices()) {
+                try {
+                    if (matchExpressionEvaluator.applies(sc.getMatchExpression(), target)) {
+                        matchedTargets.add(target);
+                    }
+                } catch (ScriptException e) {
+                    logger.error(e);
+                    break;
+                }
+            }
+            return matchedTargets;
+        }
+    }
+
+    public void delete(int id) throws IOException {
+        Path path = credentialsDir.resolve(String.valueOf(id));
+        if (!fs.isRegularFile(path)) {
+            throw new FileNotFoundException();
+        }
+        fs.deleteIfExists(path);
+    }
+
+    public Map<Integer, String> getAll()
+            throws JsonSyntaxException, JsonIOException, NumberFormatException, IOException {
+        Map<Integer, String> result = new HashMap<>();
+
+        for (String pathString : this.fs.listDirectoryChildren(credentialsDir)) {
+            Path path = credentialsDir.resolve(pathString);
+            Path filenamePath = path.getFileName();
+            if (filenamePath == null) {
+                continue;
+            }
+            try (BufferedReader br = fs.readFile(path)) {
+                StoredCredentials sc = gson.fromJson(br, StoredCredentials.class);
+                result.put(Integer.valueOf(filenamePath.toString()), sc.getMatchExpression());
+            }
+        }
+
+        return result;
+    }
+
+    @Deprecated(since = "2.2")
+    /**
+     * @deprecated use {@link getAll} instead
+     */
+    Collection<String> getMatchExpressions()
+            throws JsonSyntaxException, JsonIOException, IOException {
+        Set<String> expressions = new HashSet<>();
+        for (String pathString : this.fs.listDirectoryChildren(credentialsDir)) {
+            Path path = credentialsDir.resolve(pathString);
+            try (BufferedReader br = fs.readFile(path)) {
+                StoredCredentials sc = gson.fromJson(br, StoredCredentials.class);
+                expressions.add(sc.getMatchExpression());
+            }
+        }
+        return expressions;
     }
 
-    public List<MatchedCredentials> getMatchExpressionsWithMatchedTargets() {
+    @Deprecated(since = "2.2")
+    /**
+     * @deprecated use {@link getAll} instead
+     */
+    public List<MatchedCredentials> getMatchExpressionsWithMatchedTargets()
```

Comment:
VSCode > "find all references" says this method is only used for unit tests. If this method isn't used in the source code anymore, why is it advantageous to mark a method as deprecated instead of deleting it and updating the unit tests?

Reply:
Good catch. I'll tackle removing that and updating tests tomorrow.

Reply:
Fixed in latest commits.

Reply:
Did you also mean to remove `getMatchExpressions()` and `getMatchExpressionsWithMatchedTargets()` or leave them annotated as deprecated?